recipes/paho-mqtt-cpp/all/conanfile.py

```diff
@@ -65,8 +66,9 @@ def _configure_cmake(self):
         return self._cmake
 
     def build(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
+        if "patches" in self.conan_data.keys():
```

Comment:
```suggestion
        if self.version in self.conan_data["patches"]:
```

patches will be present due the previous version, however, the patch version may vary.

Reply:
This was actually my first try, but due to a previous:
> [HOOK - conan-center.py] post_export(): [CONANDATA.YML REDUCE (KB-H031)] Saving conandata.yml: {'sources': {'1.1': {'sha256': 'cb0343349ed91ef51d0e76ae860d19435a730d3d355e57886bb090014cb70bbe', 'url': 'https://github.com/eclipse/paho.mqtt.cpp/archive/v1.1.tar.gz'}}}

there is actually no "patches" in `conan_data` left.

Reply:
I run into that previously as well. I believe that the shrinking of the `conandata.yml` file leaves no `patches` key behind if there are no patch files for the current version